sonarqube e nexus

thekoma opened the pull request:
sonarqube e nexus

Commit:
Lcomponents in acm

Commit:
there we go

mcaimi reviewed (changes requested):
LGTM, can you please add support for a backend database in the sonarqube deployment?
I have only added in the kubernetes "flavor" and it's not merged back into the deployment templates for OCP (look here 27024292f801e71b6997449fc9cfce6c4e5ae67b).

With this last addition I think that this can be merged. Thanks!

thekoma committed:
applications

thekoma committed:
Patching

mcaimi commented on openshift/acm-components/sonarqube/templates/deployment.yaml:
```diff
@@ -0,0 +1,84 @@
+          persistentVolumeClaim:
+            claimName: sonar-data-claim
+        - name: sonar-logs-volume
+          emptyDir: {}
```
Why an emptyDir?

mcaimi commented on experiments/jenkins.yml:
```diff
@@ -0,0 +1,33 @@
+    tag: latest
+
+
+image-registry.openshift-image-registry.svc/jenkins/jk01:latest
```
This line is bogus, a leftover I presume?

mcaimi commented on experiments/jkimage.yml:
```diff
@@ -0,0 +1,61 @@
+      - name: kubernetes-credentials-provider
+        version: '0.12.1'
+      - name: configuration-as-code-groovy
+        version: '1.1'
```
We should add all plugins needed for this demo too. The complete list is [here](https://github.com/mcaimi/jenkins/blob/main/jdk8/files/plugins-list)

mcaimi commented on openshift/cd-applications/demo-app-project.yaml:
```diff
@@ -0,0 +1,18 @@
+      kind: '*'
+  destinations:
+    - namespace: 'demo-app'
+      server: 'https://api.lab02.gpslab.club:6443'
```
Use generic placeholder urls here, or make them a variable

mcaimi commented on openshift/cd-applications/examples/acm-operator.yml:
```diff
@@ -0,0 +1,21 @@
+  namespace: argocd
+spec:
+  destination:
+    server: 'https://api.lab02.gpslab.club:6443'
```
Here too, use placeholder urls or variables

mcaimi commented on openshift/cd-applications/examples/acm-operator.yml:
```diff
@@ -0,0 +1,21 @@
+  source:
+    path: 10-acm-operator/
+    repoURL: >-
+      ssh://git@gitlab.consulting.redhat.com:2222/gps-laboratory/k8s-resources.git
```
If possible, this should be included in this public repository on github or added to another public repository.
That repo is private

mcaimi commented on openshift/cd-applications/examples/authentication.yml:
```diff
@@ -0,0 +1,21 @@
+  source:
+    path: 03-auth/
+    repoURL: >-
+      ssh://git@gitlab.consulting.redhat.com:2222/gps-laboratory/k8s-resources.git
```
Same here, move code inside this repo or into another public repo

mcaimi committed:
Do NOT push private credentials :)

Do not push private credentials to this repo, it is publicly visible on the internet

mcaimi commented on openshift/cd-applications/nexus3-app.yaml:
```diff
@@ -0,0 +1,22 @@
+    namespace: demo-app
+  source:
+    path: openshift/cd-components/nexus3
+    repoURL: 'https://github.com/thekoma/k8s-demo-app.git'
```
Refer to the parent repo if possible: we need to make sure that code is pulled from the same repository.

mcaimi reviewed (changes requested):
Can you have a look?

mcaimi reviewed (dismissed):
There are private credentals attached to this PR (4ceb2d80c85829c8126b74d212f0a42c9286ecad) Can you remove this file from the commit and update the PR?

mcaimi commented:
PR is abandoned.
